driver-core/src/main/com/mongodb/client/model/ValidationOptions.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model;
+
+import org.bson.conversions.Bson;
+
+import static com.mongodb.assertions.Assertions.notNull;
+
+/**
+ * Validation options for documents being inserted or updated in a collection
+ *
+ * @since 3.2
+ * @mongodb.server.release 3.2
+ */
+public final class ValidationOptions {
+    private Bson validator;
+    private ValidationLevel validationLevel = ValidationLevel.STRICT;
```

Comment:
We should let the server choose the default rather than select one.  It's possible that in the future the default values for level and action will be configurable.